src/projects/repeat_resolution/mdbg.hpp

```diff
@@ -0,0 +1,200 @@
+                               const UniqueClassificator &classificator);
+
+    void SpreadFrost();
+    void FreezeUnpairedVertices();
+
+    [[nodiscard]] std::unordered_map<RREdgeIndexType, Sequence>
+    GetEdgeSeqs() const;
+    [[nodiscard]] std::unordered_map<RRVertexType, Sequence>
+    GetVertexSeqs(const std::unordered_map<RREdgeIndexType, Sequence> &) const;
+
+    template<typename IndexType>
+    [[nodiscard]] std::unordered_map<IndexType, IndexType>
+    MapSeqs2RC(const std::unordered_map<IndexType, Sequence> &seqs) const;
+
+    template<typename IndexType>
+    [[nodiscard]] std::unordered_map<IndexType, bool>
+    AreSeqsCanonical(const std::unordered_map<IndexType, Sequence> &seqs) const;
+
+    void ExportToGFA(
+        const std::experimental::filesystem::path &path,
+        const std::unordered_map<RRVertexType, Sequence> &vertex_seqs,
+        const std::unordered_map<RREdgeIndexType, Sequence> &edge_seqs,
+        const std::unordered_map<RRVertexType, RRVertexType> &vertex2rc,
+        const std::unordered_map<RREdgeIndexType, RREdgeIndexType> &edge2rc,
+        const std::unordered_map<RRVertexType, bool> &vertex_can,
+        const std::unordered_map<RREdgeIndexType, bool> &edge_can) const;
+
+    [[nodiscard]] std::vector<Contig>
+    GetContigs(const std::unordered_map<RRVertexType, Sequence> &vertex_seqs,
+               const std::unordered_map<RREdgeIndexType, Sequence> &edge_seqs,
+               const std::unordered_map<RRVertexType, RRVertexType> &vertex2rc,
+               const std::unordered_map<RRVertexType, bool> &vertex_can,
+               const std::unordered_map<RREdgeIndexType, bool> &edge_can
+    ) const;
+
+    [[nodiscard]] std::vector<Contig> ExportContigs(
+        const std::experimental::filesystem::path &f,
+        const std::unordered_map<RRVertexType, Sequence> &vertex_seqs,
+        const std::unordered_map<RREdgeIndexType, Sequence> &edge_seqs,
+        const std::unordered_map<RRVertexType, RRVertexType> &vertex2rc,
+        const std::unordered_map<RRVertexType, bool> &vertex_can,
+        const std::unordered_map<RREdgeIndexType, bool> &edge_can) const;
+
+ public:
+    MultiplexDBG(const std::vector<SuccinctEdgeInfo> &edges, uint64_t start_k,
+                 RRPaths *rr_paths, bool contains_rc);
+
+    MultiplexDBG(dbg::SparseDBG &dbg, RRPaths *rr_paths, uint64_t start_k,
+                 UniqueClassificator &classificator);
+
+    MultiplexDBG(const MultiplexDBG &) = delete;
+    MultiplexDBG(MultiplexDBG &&) = default;
+    MultiplexDBG &operator=(const MultiplexDBG &) = delete;
+    MultiplexDBG &operator=(MultiplexDBG &&) = default;
+
+    void AssertValidity() const;
+
+    void ExportToDot(const std::experimental::filesystem::path &path) const;
+    void ExportToGFA(const std::experimental::filesystem::path &path) const;
+
+    [[nodiscard]] bool IsFrozen() const;
+
+    [[nodiscard]] bool IsVertexComplex(const RRVertexType &vertex) const;
```

Comment:
It is always so sad when you have to do such methods to be members of the graph rather than the vertex class itself. But I guess this is what you get when you use graph library. As a result even a single vertex method that requires graph makes it necessary to do the same for all other simple vertex property methods. I hope we switch to a single graph for all purposes at some point in the future but now there is nothing to be done about it.